### PR TITLE
Adding template for dataset request

### PR DESCRIPTION
**Pull request by NatHillardUSDS**
Part of #262 - As a developer, I want to have a YAML file describing the metadata for our existing datasets 

This is an experimental attempt to add a github form for dataset requests , which will essentially auto-validate new dataset requests at the time of input by using newly-introduced [github issue forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository). 

We can then setup actions to parse this content and do things with it, using something like [this](https://github.com/peter-murray/issue-forms-body-parser)

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/dataset_request.yaml [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    id: operating-systems
+    attributes:
+      label: What formats does this source support?
+      description: You may select more than one.
+      options:
+        - label: CSV
+        - label: XLS/XLSX
+        - label: geoJSON
+        - label: ArcGIS REST API
+        - label: other
```
Comment (ginabeena) [inline .github/ISSUE_TEMPLATE/dataset_request.yaml:54]: other...? 😈

File: .github/ISSUE_TEMPLATE/dataset_request.yaml [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    validations:
+      required: false
+  - type: dropdown
+    id: spatial-resolution
+    attributes:
+      label: What is the spatial resolution of this dataset?
+      options:
+        - County
+        - Census tract
+        - Census block group
```
Comment (ginabeena) [inline .github/ISSUE_TEMPLATE/dataset_request.yaml:70]: Wonder if people will know what these mean. Census tract vs Census block group... 🤔

File: .github/ISSUE_TEMPLATE/dataset_request.yaml [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    validations:
+      required: true
+  - type: dropdown
+    id: data-quality-rating
+    attributes:
+      label: How would you rate the quality of this data?
+      options:
+        - Low
+        - Medium
+        - High
```
Comment (ginabeena) [inline .github/ISSUE_TEMPLATE/dataset_request.yaml:97]: What would be an example of low quality data and why would someone use it?

File: .github/ISSUE_TEMPLATE/dataset_request.yaml [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+      description: Provide known margin of error if known
+      placeholder: 5%
+    validations:
+      required: false
+  - type: textarea
+    id: data-quality-issues
+    attributes:
+      label: Known data quality issues
+      description: Please describe any data quality issues you know about with this dataset.
+      placeholder: Limited accuracy in rural areas, etc.
```
Comment (ginabeena) [inline .github/ISSUE_TEMPLATE/dataset_request.yaml:113]: oh i see...

**Review by ginabeena — APPROVED** [review]
Amazing idea and I love that this hooks into forms. Great work Nat.

**Review by esfoobar-usds — APPROVED** [review]
🍏 - great work @NatHillardUSDS.

@lucasmbrown-usds: at some point we want to discuss this new approach with you.

**Comment by NatHillardUSDS** [comment]
These are all valid questions, thank you @ginabeena for weighing in, I agree with all of these. 

For now, for the purposes of discussion, our thought was to merge this as-is so we see how it looks with the github issue forms renderer, though we definitely need to tweak these going forward to made them easier to understand etc. - we can make up a followup issue for this if we decide to go forward with this setup

**Pull request merged**